docs/src/main/asciidoc/lra-guide.adoc

```diff
@@ -0,0 +1,196 @@
+
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Narayana LRA Participant Support
+
+include::./attributes.adoc[]
+
+== Introduction
+
+The LRA (short for Long Running Action) participant extension is useful in microservice based
+designs where different services can benefit from a relaxed notion of distributed consistency.
+
+The idea is for multiple services to perform different computations/actions in concert, whilst
+retaining the option to compensate for any actions performed during the computation.
+This kind of loose coupling of services bridges the gap between strong consistency models
+such as JTA/XA and "home grown" ad hoc consistency solutions.
+
+The model is based on the https://github.com/eclipse/microprofile-lra/blob/master/spec/src/main/asciidoc/microprofile-lra-spec.adoc#eclipse-microprofile-lra[Eclipse Microprofile LRA specification].
+The approach is for the developer to annotate a business method with a Java annotation
+(https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/annotation/ws/rs/LRA.java[`@LRA`]).
+When such a method is called a context is created (if one is not already present) which is passed
+along with subsequent JAX-RS invocations until a method is reached
```

Comment:
More for my understanding, is it only JAX-RS? What about gRPC and other protocols?

Reply:
Although narayana supports POJO based participants, this version of the extension only supports JAX-RS participants. The reason for this is that our narayana implementation [of MP-LRA] requires a CDI extension but these are not supported by quarkus.

If we cannot workaround this missing feature then we'll update narayana by removing the reliance on CDI extensions. Note that our initial implementation (of LRA) did not require CDI and we only added it for expediency, I think, i.e. it will be a simple exercise to strip out the CDI extension.

FYI: So the next version of the extension will support non JAX-RS participants where the participant only needs the annotations. The current narayana implementation uses a (JAX-RS) proxy to manage the interactions with the coordinator.